module/zfs/dsl_dir.c

```diff
@@ -1517,6 +1517,11 @@ dsl_dir_diduse_space(dsl_dir_t *dd, dd_used_t type,
 {
```

Comment:
It would be nice if dsl_dir_diduse_space() and dsl_dir_transfer_space() could be implemented as wrappers around dsl_dir_diduse_transfer_space(), rather than duplicating the code.

Reply:
For dsl_dir_transfer_space() it would be a functional overkill.  For dsl_dir_diduse_space() it would be good, but it would require replication of MUTEX_HELD(&dd->dd_lock) trick, which I found pretty expensive due to permanent cache miss.